decidim-assemblies/app/controllers/decidim/assemblies/assemblies_controller.rb

```diff
@@ -22,7 +22,11 @@ def index
         authorize! :read, Assembly
       end
 
-      def show; end
+      def show
+        redirect_to "/404" unless (current_participatory_space.private_space? &&
```

Comment:
can you move this to something like `current_user_can_visit_space?` so we encapsulate the logic? I feel this is hard to follow

Reply:
Is this returning a 404 Not Found HTTP status? I think we should.